scripts/copy_rename_files.py

```diff
@@ -0,0 +1,150 @@
+        :param folder_path: source folders path list
+        :param subfolder_name: subfolder name
+        '''
+        file_list = []
+        for folder_path in folder_list:
+            subfolder_path = os.path.join(folder_path, subfolder_name)
+            img_list = sort_file_list(glob(os.path.join(subfolder_path, '*.png')))
+            file_list.extend(img_list)
+        return file_list
+
+    def main(self):
+        out_train_path = os.path.join(self.des_folder, self.outfolder_list[0])
+        assert len(self.train_img_files) == len(self.train_label_files), 'training image and label not equal!'
+        num_train = len(self.train_img_files)
+
+        if self.split_flag:
+            out_valid_path = os.path.join(self.des_folder, self.outfolder_list[1])
+            assert len(self.valid_img_files) == len(self.valid_label_files), 'validation image and label not equal!'
+            num_valid = len(self.valid_img_files)
+
+        for i_train in range(num_train):
+            copy_and_rename(self.train_img_files[i_train], os.path.join(out_train_path, self.subfolder_list[0]), f'{str(i_train + 1).zfill(6)}.png')
+            copy_and_rename(self.train_label_files[i_train], os.path.join(out_train_path, self.subfolder_list[1]), f'{str(i_train + 1).zfill(6)}.png')
+
+        print('Training or Whole Data Set Copy and Rename Done')
+
+        if self.split_flag:
+            for i_valid in range(num_valid):
+                copy_and_rename(self.valid_img_files[i_valid], os.path.join(out_valid_path, self.subfolder_list[0]), f'{str(i_valid + 1).zfill(6)}.png')
+                copy_and_rename(self.valid_label_files[i_valid], os.path.join(out_valid_path, self.subfolder_list[1]), f'{str(i_valid + 1).zfill(6)}.png')
+            print('Validation Set Copy and Rename Done')
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Copy and Rename to Rearrange files')
+    parser.add_argument('-i', '--input_dir', required=False, default='../monai_data/02_AMBF_DATASETS',
+                        help='The folder of your data')
+    parser.add_argument('-o', '--output_dir', required=False, default='../monai_data/AMBF_oldtool',
+                        help='The output folder of your reorganized data')
+    parser.add_argument('-s', '--split_flag', required=False, default="True", help='The flag of whether split the dataset')
+
+    args = parser.parse_args()
+    data_path = os.path.abspath(args.input_dir)
+    out_folder = os.path.abspath(args.output_dir)
+    flag_split = json.loads(args.split_flag.lower())
+
+    # data_path = os.path.join(dynamic_path, 'monai_data', 'AMBF_DATASETS')
+    # out_folder = os.path.join(dynamic_path, 'monai_data', 'AMBF_DATASETS_New')
+
+    file_reorganizer = FilesRearrange(data_path, out_folder, flag_split)
+
+    file_reorganizer.main()
+
+    # Copy the yaml file
+    yaml_name = 'dataset_config.yaml'
+    src_yaml_file = os.path.join(dynamic_path, 'monai_data', yaml_name)
```

Comment:
This seems like something very specific to your setup. 

I would say the yaml file you be kept in the root directory of the dataset and then copied from there to the new folder.